playbooks/dependencies.yaml

```diff
@@ -0,0 +1,44 @@
+---
+- name: Ensure k8s-cluster-api-provider project dependencies
+  hosts: all
+  vars:
+    jq_version: "1.6"
+    yq_version: "4.35.1"
+    kubectl_version: "1.28.1"
+    openstackclient_version: "6.2.0"
+    octaviaclient_version: "3.4.0"
+    tf_version: "1.4.6"
+    install_dir: "{{ ansible_user_dir }}/.local/bin"
+  environment:
+    PATH: "{{ install_dir }}:{{ ansible_env.PATH }}"
+  roles:
+  - role: ensure-pip  # https://zuul-ci.org/docs/zuul-jobs/latest/python-roles.html#role-ensure-pip
+  - role: ensure-terraform  # https://zuul-ci.org/docs/zuul-jobs/latest/hashicorp-roles.html#role-ensure-terraform
+    vars:
+      terraform_version: "{{ tf_version }}"
+      terraform_install_dir: "{{ install_dir }}"
+  tasks:
+  - name: Install jq
+    get_url:
+      url: "https://github.com/jqlang/jq/releases/download/jq-{{ jq_version }}/jq-linux64"
+      dest: "{{ install_dir }}/jq"
+      mode: "+x"
+      # TODO: use `checksum` attr here to verify the digest of the destination file, if available
+  - name: Install yq
+    get_url:
+      url: "https://github.com/mikefarah/yq/releases/download/v{{ yq_version }}/yq_linux_amd64"
+      dest: "{{ install_dir }}/yq"
+      mode: "+x"
+      # TODO: use `checksum` attr here to verify the digest of the destination file, if available
+  - name: Install kubectl
+    get_url:
+      url: "https://dl.k8s.io/release/v{{ kubectl_version }}/bin/linux/amd64/kubectl"
+      dest: "{{ install_dir }}/kubectl"
+      mode: "+x"
+      # TODO: use `checksum` attr here to verify the digest of the destination file, if available
```

Comment:
I think it makes sense to have a ensure-kubectl role/job in the future.

Reply:
Good idea, this will addressed within https://github.com/SovereignCloudStack/k8s-cluster-api-provider/issues/568